packages/graphql/letter-to-santa/model-property-error-handling.md

```diff
@@ -0,0 +1,1303 @@
+# Proposal: Expressing Model Property Error Handling
+
+This proposal suggests adding two new decorators to the TypeSpec standard library:
+
+- `@throws` decorator: This decorator is used to specify that the use of a model property may result in specific errors being produced.
+- `@handles` decorator: This decorator is used to specify that an operation or property will handle certain types of errors, preventing them from being propagated.
+
+It also proposes updating existing emitters to support these new decorators.
+
+<br>
+
+## Goals
+
+1. Provide a way for TSP developers to document errors associated with a particular model property.
+2. Provide spec emitters with information that can be used to update the set of operation errors based on the models in use, and the error handling of the operation.
+3. Allow code emitters to generate code that expects and handles these errors appropriately.
+
+<br>
+
+## Definitions
+
+### `@throws` decorator
+
+````typespec
+/**
+ * Specify that the use of this property may result in specific errors being produced.
+ *
+ * @param errors The list of error models that may be produced when using this property.
+ *
+ * @example
+ *
+ * ```typespec
+ * model User {
+ *   @throws(NotFoundError, PermissionDeniedError, InvalidURLError)
```

Comment:
Are these going to be decorated with `@error`? It would be good to show that in order to be explicit about it.

Reply:
From what I've seen in TypeSpec code, the `@example`s that are included in the doc blocks of decorators are very short and sweet. I think we're covering it sufficiently with (emphasis mine)
> The list of **error models** that may be produced when using this property.